### ISSUE TITLE
Inquiry on syntax highlighting in the web app

### ISSUE THREAD

**Issue by Neonnaut** [def-gthill/lexurgy#79]
Hi Graham.

I've been making my own syntax highlighter for something called Lexifer. [neonnaut.github.io/lexifer.html](https://neonnaut.github.io/lexifer.html). Even though I have achieved this by placing a div over the textbox using CSS grid, I noticed that your web app [lexurgy-app.vercel.app/sc](https://lexurgy-app.vercel.app/sc) doesn't do this. Your web app uses a contentedible div instead.

But when I tried using a contentedible div, I ran into two big problems. Text that was pasted into the div retained its formatting, and the caret would jump to the end of the text in the div. Your web app doesn't do this.

How did you achieve this?

**Comment by def-gthill** [comment]
I used [CodeMirror](https://codemirror.net/docs/). I have no idea how they get it all to work so smoothly, but thankfully I don't have to, someone's already figured it all out.

**Comment by Neonnaut** [comment]
How did you add your own language module?

**Comment by def-gthill** [comment]
https://codemirror.net/examples/lang-package/

**Comment by Neonnaut** [comment]
Okay I know this isn't the place to ask this, but I'm having trouble writing a .grammar doc. I have comments and numbers figured out. I know there are the docs, examples and [lezer-playground.vercel.app](https://lezer-playground.vercel.app/), but they're all on very complicated languages, or very basic contextless languages to figure out how mine would work on a basic level.

If only I had a small template that did the following: 

A set of functions that are highlighted at the beginning of a line disregarding whitespace, and preceding an indent
A set of keywords that are highlighted following a function on the same line
A set of characters (";" and "+"), that are highlighted following a function on the same line
```
   function1: keyword1 kyword2
function2: ab;c+d
```

**Comment by def-gthill** [comment]
You're correct that this isn't the place to ask. If you have specific questions about why something isn't working the way you expect, I can try to answer them.
